### PR TITLE
adds services

### PR DESCRIPTION
**Pull request by rafaparadela**
This PR adds Profotobuf protocol to be able to infer the services described in the proto files.

### CODE REVIEW

File: src/main/scala/higherkindness/skeuomorph/mu/protocol.scala [hunk 1]
```diff
@@ -59,6 +62,29 @@ object Protocol {
       List(Service(proto.name, SerializationType.Avro, proto.messages.map(toOperation)))
     )
   }
+
+  def fromProtobufProto[T, U](
+      protocol: protobuf.Protocol[T])(implicit T: Basis[ProtobufF, T], U: Basis[MuF, U]): Protocol[U] = {
+    val toFreestyle: T => U = scheme.cata(transformProto[U].algebra)
```
Comment (juanpedromoreno) [inline src/main/scala/higherkindness/skeuomorph/mu/protocol.scala]: toMu

File: src/main/scala/higherkindness/skeuomorph/protobuf/Protocol.scala [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        },
+        Nil
+      )
+  }
+
+  def fromFreesFProtocol[T, U](
+      protocol: mu.Protocol[T])(implicit T: Basis[MuF, T], U: Basis[ProtobufF, U]): Protocol[U] = {
+    import higherkindness.skeuomorph.mu.Service.{Operation => MuOperation}
+
+    def fromFreestyle: T => U = scheme.cata(fromFreesFSchema.algebra)
```
Comment (juanpedromoreno) [inline src/main/scala/higherkindness/skeuomorph/protobuf/Protocol.scala]: fromMu

File: src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala [hunk 3]
```diff
@@ -100,128 +102,134 @@ final case class NativeOption(name: String, value: String)
...
+          isRepeated = true,
+          isMapField = false
+        )
+      case (_, true) =>
+        NativeField(
+          name = field.getName,
+          position = field.getNumber,
+          tpe = getNativeMap(field.getName, source, files),
+          options = fromFieldOptionsMsg(field.getOptions),
+          isRepeated = false,
```
Comment (juanpedromoreno) [inline src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala]: Is this false? shouldn't be `field.getLabel.isRepeated`?

File: src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala [hunk 4]
```diff
@@ -100,128 +102,134 @@ final case class NativeOption(name: String, value: String)
...
+    case (oneof, index) => {
+      val nativeFields: List[NativeField] =
+        fields
+          .filter(t => t.hasOneofIndex && t.getOneofIndex == index)
+          .map(fromFieldDescriptorProto(_, source, files))
+          .collect { case b: NativeField => b }
+
+      NativeOneOfField(
+        name = oneof.getName,
+        tpe = NativeOneOf(name = oneof.getName, fields = NonEmptyList(nativeFields.head, nativeFields.tail)))
```
Comment (juanpedromoreno) [inline src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala]: nativeFields.head 🔥 

That wouldn't be safe enough

File: src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala [hunk 5]
```diff
@@ -100,128 +102,134 @@ final case class NativeOption(name: String, value: String)
...
-      val separateValueFromAliases = hasAlias.map(list => (list.head, list.tail))
-      (
-        (separateValueFromAliases.map(_._1) ++ noAlias.flatten).toList
-          .sortBy(_.getNumber)
-          .map(i => (i.getName, i.getNumber)),
-        separateValueFromAliases.flatMap(_._2).toList.map(i => (i.getName, i.getNumber)))
+  def partitionValuesAliases(
+      valuesAndAliases: List[EnumValueDescriptorProto]): (List[(String, Int)], List[(String, Int)]) = {
+    val (hasAlias, noAlias)      = valuesAndAliases.groupBy(_.getNumber).values.partition(_.lengthCompare(1) > 0)
+    val separateValueFromAliases = hasAlias.map(list => (list.head, list.tail))
```
Comment (juanpedromoreno) [inline src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala]: same here `.head`

File: src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala (hunk 3 above)
Comment (rafaparadela) [reply src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala]: This particular `NativeField` includes the `NativeMap` in the `tpe` whose structure represents the iterativity per se. `isRepeated` is used here to wrap this field with a list, which is not necessary in the case of the maps. More info here: https://developers.google.com/protocol-buffers/docs/proto#maps (`Maps cannot be repeated`)

**Comment by rafaparadela** [comment]
Closed in favor of https://github.com/higherkindness/skeuomorph/pull/66

File: src/main/scala/higherkindness/skeuomorph/mu/protocol.scala (hunk 1 above)
Comment (rafaparadela) [reply src/main/scala/higherkindness/skeuomorph/mu/protocol.scala]: Solved in https://github.com/higherkindness/skeuomorph/pull/66/files#diff-f3d844699128986e9c24c8908871ee30R68

File: src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala (hunk 4 above)
Comment (rafaparadela) [reply src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala]: Solved at https://github.com/higherkindness/skeuomorph/pull/66/files#diff-f159469eb4fabafc5eab8087aa5e3619R273

File: src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala (hunk 5 above)
Comment (rafaparadela) [reply src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala]: Solved at https://github.com/higherkindness/skeuomorph/pull/66/files#diff-f159469eb4fabafc5eab8087aa5e3619R189

**Pull request closed without merging**